### PR TITLE
bls sig checker compiles

### PR DESCRIPTION
**Pull request by gpsanant**
- moved the BLSSigChecker to the new registry architechture

### CODE REVIEW

File: src/contracts/middleware/BLSSignatureChecker.sol [hunk 1]
```diff
@@ -92,24 +107,34 @@ abstract contract BLSSignatureChecker {
         {
             // the quorumBitmaps of the nonSigners
             uint256[] memory nonSignerQuorumBitmaps = new uint256[](nonSignerStakesAndSignature.nonSignerPubkeys.length);
+            {
+                // the bitmap of the quorumNumbers
+                uint256 singingQuorumBitmap = BytesArrayBitmaps.bytesArrayToBitmap(quorumNumbers);
```
Comment (Sidu28) [inline src/contracts/middleware/BLSSignatureChecker.sol:112]: not quite understanding this variable name

Comment (gpsanant) [reply src/contracts/middleware/BLSSignatureChecker.sol]: haha fixed: https://github.com/Layr-Labs/eigenlayer-contracts/pull/74/commits/cbadd96f524f26254528ece2a95645c55c3edeb8

File: src/contracts/middleware/BLSSignatureChecker.sol [hunk 2]
```diff
@@ -92,24 +107,34 @@ abstract contract BLSSignatureChecker {
...
-                nonSignerPubkeyHashes[i] = nonSignerStakesAndSignature.nonSignerPubkeys[i].hashG1Point();
-                if (i != 0) {
-                    require(uint256(nonSignerPubkeyHashes[i]) > uint256(nonSignerPubkeyHashes[i - 1]), "BLSSignatureChecker.checkSignatures: nonSignerPubkeys not sorted");
+                for (uint i = 0; i < nonSignerStakesAndSignature.nonSignerPubkeys.length; i++) {
+                    nonSignerPubkeyHashes[i] = nonSignerStakesAndSignature.nonSignerPubkeys[i].hashG1Point();
+                    if (i != 0) {
+                        require(uint256(nonSignerPubkeyHashes[i]) > uint256(nonSignerPubkeyHashes[i - 1]), "BLSSignatureChecker.checkSignatures: nonSignerPubkeys not sorted");
+                    }
+                    nonSignerQuorumBitmaps[i] = registryCoordinator.operatorIdToQuorumBitmap(nonSignerPubkeyHashes[i]);
+                    // subtract the nonSignerPubkey from the running apk to get the apk of all signers
```
Comment (Sidu28) [inline src/contracts/middleware/BLSSignatureChecker.sol:120]: maybe we can add a more descriptive comment here

File: src/contracts/middleware/BLSIndexRegistryCoordinator.sol [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
-import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IBLSStakeRegistryCoordinator.sol";
 import "../interfaces/IIndexRegistry.sol";
 
 import "../libraries/BytesArrayBitmaps.sol";
 
 import "./StakeRegistry.sol";
 
-contract BLSIndexRegistryCoordinator is StakeRegistry, IRegistryCoordinator {
+contract BLSIndexRegistryCoordinator is StakeRegistry, IBLSStakeRegistryCoordinator {
```
Comment (Sidu28) [inline src/contracts/middleware/BLSIndexRegistryCoordinator.sol:11]: This naming scheme doesn't make sense to me.  BLSRegistryCoordinator seems to be more appropriate.  Same goes for the interface....

File: src/contracts/interfaces/IBLSStakeRegistryCoordinator.sol [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * @title Minimal interface for the `IBLSStakeRegistryCoordinator` contract.
+ * @author Layr Labs, Inc.
+ */
+interface IBLSStakeRegistryCoordinator is IRegistryCoordinator {
+    /// @notice the stake registry for this corrdinator is the contract itself
+    function stakeRegistry() external view returns (IStakeRegistry);
+    /// @notice the BLS Pubkey Registry contract that will keep track of operators' BLS public keys
+    function blsPubkeyRegistry() external view returns (IBLSPubkeyRegistry);
```
Comment (Sidu28) [inline src/contracts/interfaces/IBLSStakeRegistryCoordinator.sol:16]: Perhaps I'm missing something here, why don't we define an indexRegistry getter here?

**Pull request merged**